src/dash-table/components/ControlledTable/index.tsx

```diff
@@ -183,8 +184,23 @@ export default class ControlledTable extends PureComponent<ControlledTableProps>
              * so, only call when the table isn't already focussed, otherwise
              * the app will excessively re-render on _every click on the page_
              */
-            this.props.is_focused) {
-            this.props.setProps({ is_focused: false });
+            this.props.is_focused
+        ) {
+            this.props.setProps({
+                is_focused: false
+            });
+        }
+
+        const menu = this.menuRef;
+
+        if (this.props.activeMenu &&
+            menu &&
+            menu.current &&
+            !menu.current.contains(event.target as Node)
+        ) {
+            this.props.setState({
+                activeMenu: undefined
+            });
```

Comment:
Generalized the `click-outside` handler. Now checks for two different / unrelated cases. (a) table's lost of focus, (b) closing the columns menu

Reply:
Using the new way of `ref` in React 16

Reply:
neat, wasn't aware of `React.createRef` ✨

Reply:
Do we want menu clicks to drop focus on the table? Looks like that's what this will do, but feels to me like the focus should remain.

Reply:
The focus remains. If you click to open the menu, you can still type / navigate the table and the `focused` cell highlight isn't lost. `onBlur` will happen though as you clicked outside the current cell.